Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp

```diff
@@ -519,10 +519,10 @@ void BytecodeInterpreter::interpret(Configuration& configuration, InstructionPoi
         configuration.stack().push(Value(ValueType { ValueType::I64 }, instruction.arguments().get<i64>()));
         return;
     case Instructions::f32_const.value():
-        configuration.stack().push(Value(ValueType { ValueType::F32 }, static_cast<double>(instruction.arguments().get<float>())));
+        configuration.stack().push(Value(Value::AnyValueType(instruction.arguments().get<float>())));
```

Comment:
This might be a little hacky/weird, but I'm doing this because `static_cast`-ing from `double` to `float` is lossy and doesn't preserve NaN bits in all scenarios (I'm also not sure how platform-specific this behavior is). The previously-used `Value` constructor uses `static_cast`.

Perhaps it might be better to refactor the `Value` constructors so this pitfall isn't possible. Let me know if I should do that instead.

Reply:
This looks fine, but feel free to make the cast less jank if you want to :)